core/electron/src/test/backend/ElectronHost.test.ts

```diff
@@ -194,7 +198,7 @@ async function testWindowSizeSettings() {
   if (isXvfbRunning)
     window.emit("maximize"); // "maximize" event is not emitted when running with xvfb (linux)
   else
-    await BeDuration.wait(100); // "maximize" event is not always emitted immediately
+    await BeDuration.wait(250); // "maximize" event is not always emitted immediately
```

Comment:
Did you consider listening on the [window.on("maximize") event](https://www.electronjs.org/docs/latest/api/browser-window/#event-maximize)? Supposedly it is emitted once maximization is complete? Promisify the event - await it, and assert.

Reply:
There is 2 problems with awaiting for events:
- Events are thrown at different times on different OS. In my testing, I noticed that on Windows event is throw immediately after calling `window.maximize` and on MacOS it takes a bit. Not that big of a problem because we can still handle it something like this:
```ts
  const maximizePromise = new Promise((resolve) => window.once("maximize", () => resolve(undefined)));
  window.maximize();
  ...
  await maximizePromise;
```
- Bigger issue is that calling something like `window.setSize` immediately after "maximize" or "unmaximize" event is thrown, causes window to not resize (at least on Windows). For that reason, we can't replace `wait` with await for promise, only add it as addition.